core/imds/src/main/java/software/amazon/awssdk/imds/Ec2Metadata.java

```diff
@@ -29,4 +34,137 @@ public interface Ec2Metadata {
      * @return Instance metadata value
      */
     String get(String path);
+
+    /**
+     * @return The retry policy which includes the number of retry attempts for any failed request.
+     */
+    RetryPolicy retryPolicy();
+
+    /**
+     * @return The endpoint of IMDS.
+     */
+    URI endpoint();
+
+    /**
+     * @return The port for the endpoint.
+     */
+    Integer port();
+
+    /**
+     * @return The Time to live (TTL) of the token.
+     */
+    Duration tokenTtl();
+
+    /**
+     * @return The endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+     */
+    String endpointMode();
+
+    /**
+     * @return The output stream for debugging.
+     */
+    String httpDebugOutput();
+
+    /**
+     * @return The number of seconds to sleep in-between retries.
+     */
+    Integer backoff();
+
+    /**
+     * @return The SdkHttpClient instance to make the http requests.
+     */
+    SdkHttpClient httpClient();
```

Comment:
Do we need to expose getters for the settings?

Reply:
I followed the client configuration.md file , it had getters there.